pkg/operation/botanist/wanted_secrets.go

```diff
@@ -568,40 +568,46 @@ func (b *Botanist) generateWantedSecretConfigs(basicAuthAPIServer *secrets.Basic
 		})
 	}
 
-	// Secret definitions for dependency-watchdog-internal and external probes
-	secretList = append(secretList, &secrets.ControlPlaneSecretConfig{
-		CertificateSecretConfig: &secrets.CertificateSecretConfig{
-			Name: kubeapiserver.DependencyWatchdogInternalProbeSecretName,
+	if settings := b.Seed.GetInfo().Spec.Settings; settings == nil &&
+		settings.DependencyWatchdog != nil &&
+		settings.DependencyWatchdog.Probe != nil &&
+		settings.DependencyWatchdog.Probe.Enabled {
```

Comment:
This will throw a nil ptr exception if settings is nil, perhaps you meant
```suggestion
	if settings := b.Seed.GetInfo().Spec.Settings; settings != nil &&
		settings.DependencyWatchdog != nil &&
		settings.DependencyWatchdog.Probe != nil &&
		settings.DependencyWatchdog.Probe.Enabled {
```